public/static/app/perf/trend_chart.js

```diff
@@ -710,18 +710,25 @@ mciModule.factory('DrawPerfTrendChart', function (
           'stroke-width': '4',
         });
 
-      changePointSegments.select('line')
+        const changePointIndex = (index, level, series) => {
+          for (;index >= 0 && getValueFor(level)(series[index]) == null;index--) {}
```

Comment:
Could you use _.findLastIndex here? Or at least do a while loop which breaks once you've found the index. This was pretty confusing to read